stdlib/public/CTensorFlow/ctensorflow_init.cpp

```diff
@@ -34,52 +34,97 @@ void InitTensorFlowRuntime(unsigned char enable_debug_logging,
-                                      TF_Status *status) {
-  auto tfDtype = (TF_DataType)dtype;
-  auto *tensor =
-      TF_AllocateTensor(tfDtype, /*shape.data()*/ nullptr, /*shape.size()*/ 0,
-                        TF_DataTypeSize(tfDtype) * 1);
-  auto *ptr = reinterpret_cast<char *>(TF_TensorData(tensor));
-
+static bool setValue(TF_DataType tfDtype, int64_t val, void *ptr) {
   switch (tfDtype) {
   case TF_INT8:
     *reinterpret_cast<int8_t *>(ptr) = static_cast<int8_t>(val);
-    break;
+    return false;
   case TF_UINT8:
     *reinterpret_cast<uint8_t *>(ptr) = static_cast<uint8_t>(val);
-    break;
+    return false;
   case TF_INT16:
     *reinterpret_cast<int16_t *>(ptr) = static_cast<int16_t>(val);
-    break;
+    return false;
   case TF_UINT16:
     *reinterpret_cast<uint16_t *>(ptr) = static_cast<uint16_t>(val);
-    break;
+    return false;
   case TF_INT32:
     *reinterpret_cast<int32_t *>(ptr) = static_cast<int32_t>(val);
-    break;
+    return false;
   case TF_UINT32:
     *reinterpret_cast<uint32_t *>(ptr) = static_cast<uint32_t>(val);
-    break;
+    return false;
   case TF_INT64:
     *reinterpret_cast<int64_t *>(ptr) = static_cast<int64_t>(val);
-    break;
+    return false;
   case TF_UINT64:
     *reinterpret_cast<uint64_t *>(ptr) = static_cast<uint64_t>(val);
-    break;
+    return false;
   default:
-    TF_MakeInternalErrorStatus(status, "Unsupported data type");
+    return true;
+  }
+}
+
+void *swift_tfc_CreateIntTensor(int32_t num_dims, int64_t *dims, int64_t *vals,
+                                int32_t dtype_int, TF_Status *status) {
+  auto dtype = (TF_DataType)dtype_int;
+  auto dtypeSize = TF_DataTypeSize(dtype);
+
+  // Compute the total memory size of the tensor value.
+  // totalElements can be 0 if shape is [] (i.e., num_dims = 1, dims[0] = 0).
+  unsigned long long totalElements = 1;
+  for (int32_t i = 0; i < num_dims; ++i)
+    totalElements *= dims[i];
+
+  // Leave the code around for debugging. Can write it to TF LOG via an
+  // experimental C API if needed.
+  // printf("num_dims: %d, dtype: %d, total elts: %llu\n", num_dims, dtype_int,
+  //       totalElements);
+
+  // Make an uninitialized tensor that is big enough for our value.
+  auto *tensor =
+      TF_AllocateTensor(dtype, dims, num_dims, dtypeSize * totalElements);
+
+  // Set up its contents, element-wise.
+  // FIXME: This will need a byte swap for big endian hosts.
+  auto *ptr = (char *)TF_TensorData(tensor);
+  for (unsigned long long i = 0; i < totalElements; ++i) {
+    if (setValue(dtype, vals[i], ptr)) {
+      TF_MakeInternalErrorStatus(status, "Unsupported data type");
+      return nullptr;
+    }
+    ptr += dtypeSize;
+  }
+  return tensor;
+}
+
+void *swift_tfc_CreateFloatTensor(int32_t num_dims, int64_t *dims, float *vals,
+                                  TF_Status *status) {
+  // Compute the total memory size of the tensor value.
+  unsigned long long totalElements = 1;
```

Comment:
[TensorFlow's tensor byte size](https://github.com/tensorflow/tensorflow/blob/3b94d75a9e10ef8ef33760d0ef6aad326e1353ba/tensorflow/c/c_api.h#L270) representation is `size_t`, not ULL. They are often different I think.